.github/workflows/ci_macos.yaml

```diff
@@ -24,6 +24,7 @@ jobs:
     env:
       boost_path: "${{ github.workspace }}/../boost"
       openssl_root: /usr/local/opt/openssl@1.1
+      harden_flags: "-D_FORTIFY_SOURCE=2"
```

Comment:
Is GNU C Library used on macOS?

Reply:
no, it is clang and its c library AFAIK.

Reply:
IIRC, `_FORTIFY_SOURCE` is feature of GNU C Library.

Reply:
>### Fortification of function calls
This section contains _implementation details of the GNU C Library_ and may not remain stable across releases.
>
>The `_FORTIFY_SOURCE` macro may be defined by users to control hardening of calls into some _functions in the GNU C Library_.

Reply:
removed.